website/source/docs/internals/anti-entropy.html.markdown

```diff
@@ -0,0 +1,131 @@
+---
+layout: "docs"
+page_title: "Anti-Entropy"
+sidebar_current: "docs-internals-anti-entropy"
+description: >
+  This section details the process and use of anti-entropy in Consul.
+---
+
+# Anti-Entropy
+
+Consul uses an advanced method of maintaining service and health information.
+This page details how services and checks are registered, how the catalog is
+populated, and how health status information is updated as it changes.
+
+~> **Advanced Topic!** This page covers technical details of
+the internals of Consul. You don't need to know these details to effectively
+operate and use Consul. These details are documented here for those who wish
+to learn about them without having to go spelunking through the source code.
+
+### Components
+
+It is important to first understand the moving pieces involved in services and
+health checks: the [agent](#agent) and the [catalog](#catalog). These are
+described conceptually below to make anti-entropy easier to understand.
+
+<a name="agent"></a>
+#### Agent
+
+Each Consul agent maintains its own set of service and check registrations, as
+well as health information. The agents are responsible for executing their own
+health checks and updating their local state.
+
+Services and checks within the context of an agent have a rich set of
+configuration options available. This is because the agent is responsible for
+generating information about its services and their health through the use of
+[health checks](/docs/agent/checks.html).
+
+<a name="catalog"></a>
+#### Catalog
+
+Consul's service discovery is backed by a service catalog. This catalog is
+formed by aggregating information submitted by the agents. The catalog maintains
+the high-level view of the cluster, including which services are available,
+which nodes run those services, health information, and more. The catalog is
+used to expose this information via the various interfaces Consul provides,
+including DNS and HTTP.
+
+Services and checks within the context of the catalog have a much more limited
+set of fields when compared with the agent. This is because the catalog is only
+responsible for recording and returning information *about* services, nodes, and
+health.
+
+The catalog is maintained only by server nodes. This is because the catalog is
+replicated via the [Raft log](/docs/internals/consensus.html) to provide a
+consolidated and consistent view of the cluster.
+
+<a name="anti-entropy"></a>
+### Anti-Entropy
+
+Now that we have covered the functions of the agent and the catalog, we
+need a mechanism to perform synchronization between the two. In Consul, this
+method is known as anti-entropy.
```

Comment:
Consider: "Consul has a clear separation between the global service catalog and the agent local state as discussed above. Reconciling these two is done using an anti-entropy mechanism."